drivers/dai/nxp/sai/sai.c

```diff
@@ -886,6 +886,21 @@ static int sai_init(const struct device *dev)
 
 	device_map(&data->regmap, cfg->regmap_phys, cfg->regmap_size, K_MEM_CACHE_NONE);
 
+	if (cfg->dline_count <= 0) {
+		LOG_ERR("Invalid number of data lines");
+		return -EINVAL;
+	}
+
+	if (cfg->tx_dline >= cfg->dline_count) {
```

Comment:
how about:

```
#define SAI_DLINE_COUNT(base) FSL_FEATURE_SAI_CHANNEL_COUNTn(UINT_TO_I2S(base))
```

and

```
if (cfg->tx_dline >= SAI_DLINE_COUNT(cfg->regmap_phys)) {
    /* your code goes here */
}
```

I would like to avoid wasting memory for a variable that's only used in validation.

I also think @decsny's suggestion would work just fine, but then we'd end up with a new DT property whose sole purpose would be to help validate the values passed through the `tx_dataline` and `rx_dataline` DT properties.